### PR TITLE
change pos.tagset for IULA Treebank model

### PR DESCRIPTION
**Pull request by jgrivolla**
The pos-tagset used in the IULA Treebank (and which is used also by the Freeling tool) is actually the same as the one in the Ancora corpus. The dependency tagset however is different.

### CODE REVIEW

File: dkpro-core-maltparser-asl/src/scripts/build.xml [hunk 1]
```diff
@@ -309,8 +309,8 @@
 				<entry key="DC.creator" value="Universitat Pompeu Fabra. Institut Universitari de Lingüística Aplicada (IULA)"/>
 				<entry key="DC.identifier" value="http://hdl.handle.net/10230/20407"/>
 				<entry key="DC.rights" value="http://creativecommons.org/licenses/by/3.0/"/>
-	    		<entry key="pos.tagset" value="freeling"/>
-				<entry key="dependency.tagset" value="iula"/>
+	    		<entry key="pos.tagset" value="ancora"/>
```
Comment (reckart) [inline dkpro-core-maltparser-asl/src/scripts/build.xml]: When changing the metadata, the metadata version needs to be increased as well - here and also in the respective dependency in the pom.xml.

**Comment by jgrivolla** [comment]
Oops, hold off on the pull request. I'm getting conflicting information about this, so I'll keep reading and asking around to make sure.

**Comment by reckart** [comment]
@jgrivolla Thanks for spotting this.

Changing the tagset from "freeling" to "ancora" link the model to [es-ancora-pos.map](https://github.com/dkpro/dkpro-core/blob/master/dkpro-core-api-lexmorph-asl/src/main/resources/de/tudarmstadt/ukp/dkpro/core/api/lexmorph/tagset/es-ancora-pos.map). However, since POS is an input layer for MaltParser, that doesn't have much of a practical effect outside of the documentation.

If you look at the POS tagset extracted from the model in the [Spanish MaltParserTest](https://github.com/dkpro/dkpro-core/blob/master/dkpro-core-maltparser-asl/src/test/java/de/tudarmstadt/ukp/dkpro/core/maltparser/MaltParserTest.java#L410), you see that there are some differences to the `es-ancora-pos.map`. The most obvious one is that in the map all the labels are lowercase and in the model they are mostly uppercase.

You may also note a comment in the unit test, that we presently don't have a POS tagger that would create the required input tags to match what MaltParser expects. Do you have some pointer to a component/model that could be integrated into DKPro Core?

**Comment by reckart** [comment]
Please create an issue before opening a pull request. 

In the commit comments of the PR, please reference the issue as follows:

```
#<issue number> - <issue title>
- <change 1>
- <change 2>
- ...
```

**Comment by jgrivolla** [comment]
I'm working on getting fully compatible end-to-end pipelines for Spanish and hope to be able to contribute something soon. The only component I know right now for sure that produces the required input for the IULA model is [Freeling](http://nlp.cs.upc.edu/freeling/), but it's written in C and under the Affero GPL license.

I'll create an issue first next time.

**Comment by reckart** [comment]
If there is a suitable dataset for training - there is an OpenNlpPosTaggerTrainer component in DKPro Core ;)

**Comment by jgrivolla** [comment]
The problem is that Freeling is rule-based, so we can't just use the same training corpus (since it does not exist). We could of course annotate a corpus using Freeling and train on that, but that's not ideal.

**Comment by jgrivolla** [comment]
Then there's the IXA models, which should be standard Ancora (https://github.com/ixa-ehu/ixa-pipe-pos/issues/7) but actually produce uppercase tags: https://github.com/dkpro/dkpro-core/blob/master/dkpro-core-opennlp-asl/src/test/java/de/tudarmstadt/ukp/dkpro/core/opennlp/OpenNlpPosTaggerTest.java#L217

These might actually be quite close to the Freeling ones, but there's no way to really know, even when speaking to the people involved...

**Comment by reckart** [comment]
Right... the IXA models came in long after the MaltParser model. Would be worth changing the MaltParser test to use them.

**Comment by ragerri** [comment]
Just to add to the noise, IXA pipes 1.1.1 for Spanish provides: tok, pos, lemma, ner and constituent parsing. Everything supervised trained with Ancora but the tokenizer. IXA pipes currently does NOT distribute a dependency parser. If you update to 1.1.1 models and have any tests they will break.

http://ixa2.si.ehu.es/ixa-pipes/

I do not modify the annotations for any of the tasks. For NER I distribute 6 class models, but I have also models trained with 4 classes.

**Comment by reckart** [comment]
@ragerri thanks for the input. While we're at it ;) ... some time back I tried to train models from Ancora for a tech experiment. I noticed that Ancora doesn't have POS tags for all tokens. Stanford CoreNLP uses a heuristic to reconstruct the missing tags before passing them on to the trainer. How do you handle that?

**Comment by ragerri** [comment]
What do you mean Ancora does not have POS tags for all the tokens? The joint expressions such as 

26_de_mayo?

I do not treat them and I do not like the fact that the original text is not available, just as it is in the UD corpora. I let the sequence labeler to learn them (26 number, de  prep and mayo noun) from the tokens that are actual tokens.

**Comment by reckart** [comment]
@ragerri I guess I probably mean those. I didn't entirely understand the last sentence of your comment though:

> I let the sequence labeler to learn them (26 number, de prep and mayo noun) from the tokens that are actual tokens.

You mean *if* the tokens are stored as separate tokens with POS tags, then you learn from them. But you have a composite token like `a_a` (not sure if that exists in Ancora, but e.g. in the Portuguese Floresta) then you do what? You learn that token with its POS tag or you ignore sentences that contain such tokens?

Btw. did you know that Ancora is ridden with invisible unicode characters that should be removed before training a model?

**Comment by jgrivolla** [comment]
@reckart Where did you get the Stanford models (https://dkpro.github.io/dkpro-core/releases/1.8.0/docs/model-reference.html#_stanfordparser_3_2270), there's no source? It seems that AnCora actually does have upper-case POS-tags, but it's not even consistent in the original paper.

**Comment by reckart** [comment]
The source is in the build.xml file of the `stanfordnlp-gpl` module. But the build.xml file is not written in the way that the documentation generation code expects it. It expects that each model is downloaded from one URL, but Stanford packages multiple models in an archive. Thus the source does not appear in the documentation.

http://nlp.stanford.edu/software/stanford-spanish-corenlp-2016-10-31-models.jar

**Comment by ragerri** [comment]
@reckart :"Btw. did you know that Ancora is full of with invisible unicode characters that should be removed before training a model?"

Can you elaborate?

With respect to your previous question, I do not train on "26_de_mayo", I ignore it.

**Comment by reckart** [comment]
When I found that some OpenNLP pre-trained models were trained with broken encodings, I started looking around for potential problems in other pre-trained models. I found that models trained on Ancora contain SHY (soft-hypen) characters. Normal texts do not contain SHY characters. I expect that the Spanish models would perform somewhat better if the SHY characters were removed from the tokens before training. Up to 4.4% of the words in the respective model dictionaries are affected.

See here: http://langrid.org/wlsi-oiaf4hlt-2016/book.pdf?page=19

**Comment by ragerri** [comment]
OK, I read the paper, interesting. I see that this was motivated by the Stanford Spanish components in which they do modify the tagsets without actually releasing the modified corpus or the programs to do the modifications. I asked them last year and they did not release it. I checked my formatted Ancora corpora and I do not have any non ASCII characters that are not the 8859-1 required for accents and the like.

**Comment by reckart** [comment]
@ragerri The code to infer POS-tags over multi-word-tokens is part of CoreNLP: https://github.com/stanfordnlp/CoreNLP/blob/master/src/edu/stanford/nlp/international/spanish/pipeline/MultiWordPreprocessor.java

Hm. I wonder if the Spanish models are partially trained on the original CoNLL 2009 data since the OpenNLP NER models trained on that dataset are also full of SHY characters.

**Comment by ragerri** [comment]
https://mailman.stanford.edu/pipermail/java-nlp-user/2015-November/007323.html

**Comment by ragerri** [comment]
Going back to the point, we did not modify Ancora tagset, just used it "as is". For the record :)
